components/admin_console/s3_connection_test/s3_connection_test.jsx

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) 2018-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import {FormattedMessage} from 'react-intl';
+
+import * as Utils from 'utils/utils.jsx';
+
+export default class S3ConnectionTestButton extends React.Component {
+    static get propTypes() {
+        return {
+            config: PropTypes.object.isRequired,
+            getConfigFromState: PropTypes.func.isRequired,
+            disabled: PropTypes.bool.isRequired,
+            actions: PropTypes.shape({
+                testS3Connection: PropTypes.func.isRequired
+            }).isRequired
+        };
+    }
+
+    constructor(props) {
+        super(props);
+
+        this.handleTestConnection = this.handleTestConnection.bind(this);
```

Comment:
As a convention, please remove this and just use arrow function, like:
```
handleTestConnection = (e) => {...}
```

Reply:
done